### PR TITLE
fix: PATH for zsh

### PR DESCRIPTION
**Pull request by timechess**

### CODE REVIEW

**Review by alissa-tung — CHANGES_REQUESTED** [review]
Thank you! here are some suggestions

**Comment by alissa-tung** [comment]
You can test the changes by using `go build main.go` in the project root directory

File: glean/elan.go [hunk 1]
```diff
@@ -43,7 +43,8 @@ func InstallElan() {
 	case "windows":
 		cmd = exec.Command("powershell", "-ExecutionPolicy", "Bypass", "-f", scriptPath)
 	default:
-		cmd = exec.Command("/bin/sh", scriptPath, "-y", "--default-toolchain", "none")
+		cmd = exec.Command("/bin/sh", scriptPath, "-y", "--default-toolchain", "none", "&& echo PATH=\"$HOME/.elan/bin:$PATH\" >> $HOME/.zprofile")
+
```
[the remark this answers is not in the document]
Comment (alissa-tung) [reply glean/elan.go:47]: And you can also add some thing like

```sh
########
# begin @generated by glean
# end @generated by glean
########
```

**Comment by timechess** [comment]
Done it. It works on my Ubuntu. Hopefully it will also work on Mac.

File: glean/elan.go [hunk 2]
```diff
@@ -44,7 +44,23 @@ func InstallElan() {
 		cmd = exec.Command("powershell", "-ExecutionPolicy", "Bypass", "-f", scriptPath)
 	default:
 		cmd = exec.Command("/bin/sh", scriptPath, "-y", "--default-toolchain", "none")
+
+		zshText := ` '########
+# begin @generated by glean
+PATH="$HOME/.elan/bin:$PATH"
+# end @generated by glean
+########'`
+		zpfPath := os.Getenv("HOME") + "/.zprofile"
```
Comment (alissa-tung) [inline glean/elan.go:53]: Could you please use join for paths here?

File: glean/elan.go [hunk 3]
```diff
@@ -44,7 +44,23 @@ func InstallElan() {
...
+		zshcmd := exec.Command("/bin/sh","-c", "echo" + zshText+ ">>" + zpfPath)
```
Comment (alissa-tung) [inline glean/elan.go:54]: It would be easier to just use append method for file

**Review by alissa-tung — CHANGES_REQUESTED** [review]

**Pull request merged**